nix/weaviate.nix

```diff
@@ -0,0 +1,102 @@
+{ pkgs, lib, name, config, ... }:
+let
+  inherit (lib) types;
+in
+{
+  options = {
+    enable = lib.mkEnableOption name;
+
+    package = lib.mkPackageOption pkgs "weaviate" { };
+
+    dataDir = lib.mkOption {
+      type = types.str;
+      default = "./data";
+      description = "Path to the Weaviate data store";
+    };
+
+    host = lib.mkOption {
+      type = types.nullOr types.str;
+      default = "127.0.0.1";
+      description = ''
+        The IP to listen on
+      '';
+      example = "0.0.0.0";
+    };
+
+    port = lib.mkOption {
+      type = types.port;
+      default = 8080;
+      description = ''
+        The port to listen on for connections
+      '';
+    };
+
+    envs = lib.mkOption {
+      type = types.attrsOf (types.oneOf [ types.str types.int types.bool (types.listOf types.str) ]);
+      default = { };
+      description = ''
+        Weaviate environment variables.
+      '';
+      example = lib.literalExpression ''
+        {
+          AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED = true;
+          QUERY_DEFAULTS_LIMIT = 100;
+          DISABLE_TELEMETRY = true;
+          LIMIT_RESOURCES = true;
+          ENABLE_MODULES = ["text2vec-openai" "generative-openai"];
+        }
+      '';
+    };
+
+    outputs.settings = lib.mkOption {
+      type = types.deferredModule;
+      internal = true;
+      readOnly = true;
+      default = {
+        processes = {
+          "${name}" =
+            let
+              toStr = value:
+                if builtins.isString value then builtins.toJSON value
+                else if builtins.isBool value then (if value then "true" else "false")
+                else if builtins.isList value then builtins.toJSON (lib.concatStringsSep "," value)
+                else if builtins.isInt value then toString value
+                else throw "Unrecognized type";
```

Comment:
Why not:

```nix
let
  asStr = value:
    if builtins.isList value then lib.concatStringsSep "," value else value;
  toStr = value:
    builtins.toJSON (asStr value);
```

?

Then, instead of:

```nix
type = types.attrsOf (types.oneOf [ types.str types.int types.bool (types.listOf types.str) ]);
```

your type can be:

```nix
type = types.attrsOf (types.oneOf [ types.raw (types.listOf types.str) ]);
```

Also, is there a better way of defining this environment type?